Fabulous.XamarinForms/src/Fabulous.XamarinForms.Core/ViewUpdaters.fs

```diff
@@ -28,6 +28,120 @@ module ViewUpdaters =
+                | _ -> [||]
+
+            // Count the existing targetColl
+            let n = prevColl.Length
+            
+            let availableKeyedElements = Dictionary<string,'T>()
+            let rest= ResizeArray<'T>()
+            let newKeys = HashSet<string>()
+            
+            for i in 0 .. coll.Length-1 do
+                 let newChild = coll.[i]
+                 let key = getKey newChild
+                 match key with
+                 | ValueSome key -> newKeys.Add key |> ignore
+                 | ValueNone _ -> ()
+                 
+            for i in 0..n-1 do
+                let key = getKey prevColl.[i]
+                match key with
+                | ValueSome key ->
+                    availableKeyedElements.[key] <- prevColl.[i]
+                | ValueNone -> rest.Add prevColl.[i]
+            
+            let contains (key:string voption) =
+                match key with
+                | ValueSome key -> availableKeyedElements.ContainsKey key
+                | ValueNone -> false
+                
+            let newKeyContains el =
+                match getKey el with
+                | ValueSome key -> newKeys.Contains key
+                | ValueNone -> false
+  
+            // Adjust the existing targetColl and create the new targetColl
+            for i in 0 .. coll.Length-1 do
+                let newChild = coll.[i]
+                let key = getKey newChild
+                let elems =
+                        seq{
+                            yield! rest
+                            yield! availableKeyedElements.Values
+                           }
+                if (prevColl|>Seq.exists (identical newChild)) then
+                    move i newChild
+                    match getKey newChild with
+                     | ValueSome key -> availableKeyedElements.Remove key |> ignore
+                     |_ ->rest.Remove newChild |> ignore
+                     
+                elif (key.IsSome)
+                then
+                  if(contains (key)) then 
+                     let previousKeyedElement = availableKeyedElements.[key.Value] // We should be safe here
+                     if(canReuse previousKeyedElement newChild) then
+                      update i previousKeyedElement newChild
+                      availableKeyedElements.Remove key.Value |> ignore
+                  else
+                     match (elems|>Seq.tryFind(fun c-> canReuse c newChild)) with
+                     | Some el ->
+                         update i el newChild
+                         match getKey el with
+                         | ValueSome key -> availableKeyedElements.Remove key |> ignore
+                         |_ ->rest.Remove el |> ignore
+                     | None ->
+                        create i newChild
+                else
+                   
+                    match (elems|>Seq.tryFind(fun c-> canReuse c newChild && (not <| newKeyContains c))) with
```

Comment:
Same here. Search inside `rest` and not `elems`.
Also `not <| newKeyContains c` has no longer any meaning with the 2nd comment I made.

Reply:
Ok